visvis2/renderers/wgpu/pointsrender.py

```diff
@@ -0,0 +1,105 @@
+import wgpu  # only for flags/enums
+from pyshader import python2shader
+from pyshader import f32, vec2, vec4
+
+from . import register_wgpu_render_function, stdinfo_uniform_type
+from ...objects import Points
+from ...materials import PointsMaterial, GaussianPointsMaterial
+
+
+@python2shader
+def vertex_shader(
+    in_pos: ("input", 0, vec4),
+    u_stdinfo: ("uniform", (0, 0), stdinfo_uniform_type),
+    u_points: ("uniform", (1, 0), PointsMaterial.uniform_type),
+    out_pos: ("output", "Position", vec4),
+    out_point_size: ("output", "PointSize", f32),
+    out_size: ("output", 0, f32),
+):
+    world_pos = u_stdinfo.world_transform * vec4(in_pos.xyz, 1.0)
+    ndc_pos = u_stdinfo.projection_transform * u_stdinfo.cam_transform * world_pos
+
+    scale_factor = u_stdinfo.physical_size.x / u_stdinfo.logical_size.x
+    size = u_points.size * scale_factor + 1.5  # plus some for aa
+
+    out_pos = ndc_pos  # noqa - shader output
+    out_point_size = size  # noqa - shader output
+    out_size = size  # noqa - shader output
+
+
+@python2shader
+def fragment_shader_points(
+    in_size: ("input", 0, f32),
+    in_point_coord: ("input", "PointCoord", vec2),
+    u_points: ("uniform", (1, 0), PointsMaterial.uniform_type),
+    out_color: ("output", 0, vec4),
+):
+    # See https://github.com/vispy/vispy/blob/master/vispy/visuals/markers.py
+    color = u_points.color
+    pcoord_pixels = in_point_coord * in_size
+    hsize = 0.5 * (in_size - 1.5)
+    aa_width = 1.0
+    d = distance(pcoord_pixels, vec2(hsize, hsize))
+    if d <= hsize - 0.5 * aa_width:
+        out_color = u_points.color.rgba  # noqa - shader output
+    elif d <= hsize + 0.5 * aa_width:
+        alpha = 0.5 + (hsize - d) / aa_width
+        alpha = alpha ** 2  # this works better
+        out_color = vec4(color.rgb, color.a * alpha)  # noqa - shader output
+    else:
+        return  # discard
```

Comment:
Look, if-statements and a builtin function to calculate the distance!

Reply:
Awesome!!